test/functional/bulk.test.js

```diff
@@ -2003,4 +2003,103 @@ describe('Bulk', function () {
         );
     });
   });
+
+  describe('Bulk operation transaction rollback', () => {
+    /** @type {import('../../src/index').MongoClient} */
+    let client;
+    /** @type {import('../../src/index').Collection<{ answer: number }>} */
+    let collection;
+
+    beforeEach(async function () {
+      const config = this.configuration;
+      client = config.newClient();
+      await client.connect();
+
+      try {
+        await client
+          .db('bulk_operation_writes_test')
+          .collection('bulk_write_transaction_test')
+          .drop();
+      } catch (_) {
+        // do not care
+      }
+
+      collection = await client
+        .db('bulk_operation_writes_test')
+        .createCollection('bulk_write_transaction_test');
+
+      await collection.deleteMany({});
+    });
+
+    it('should abort ordered/unordered bulk operation writes', {
```

Comment:
let's split up this test and the following one into 3 cases each: ordered, unordered, and mixed; and clearly identify withTransaction vs not in the name